src/finite_elasticity_routines.f90

```diff
@@ -3448,11 +3466,740 @@ SUBROUTINE FINITE_ELASTICITY_GAUSS_CAUCHY_TENSOR(EQUATIONS_SET,DEPENDENT_INTERPO
+          END DO
+
+          CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_1,N1,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(N1,F_a_inv,N1,ERR,ERROR,*999) ! F_a_inv=F_a_inv_T
+          CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_2,N2,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(N2,F_a_inv,N2,ERR,ERROR,*999) ! F_a_inv=F_a_inv_T
+          CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_3,N3,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(N3,F_a_inv,N3,ERR,ERROR,*999) ! F_a_inv=F_a_inv_T 
+
+          FREE_ENERGY=0.0_DP
+          DO i=1,3
+            FREE_ENERGY=FREE_ENERGY+C(i)/C(i+3)*( &
+              & EVALUES(1)**(C(i+3)/2.0_DP)+ &
+              & EVALUES(2)**(C(i+3)/2.0_DP)+ &
+              & EVALUES(3)**(C(i+3)/2.0_DP)-3.0_DP)
+          END DO
+          FREE_ENERGY=C(7)*FREE_ENERGY
+
+          VALUE=XB_ENERGY_PER_VOLUME-(FREE_ENERGY-FREE_ENERGY_0)
+        ENDIF
+      ENDDO
+    
+      ! Neo-Hook
+!      F_a = 0.0_DP
+!      F_a(1,1) = lambda_a
+!      F_a(2,2) = 1.0_DP
+!      F_a(3,3) = 1.0_DP
+!      CALL MATRIX_TRANSPOSE(F_a,F_a_T,ERR,ERROR,*999)
+!      CALL MATRIX_PRODUCT(F_a_T,F_a,C_a,ERR,ERROR,*999)
+!      CALL INVERT(C_a,C_a_inv,a,ERR,ERROR,*999) !a is not required (=1/lambda_a^2 ?)
+!      PIOLA_TENSOR=C(1)*C_a_inv+2.0_DP*P*AZU
+
+      !Odgen
+!      CALL Eigenvector(C_e,EVALUES(1),EVECTOR_1,ERR,ERROR,*999)
+!      CALL Eigenvector(C_e,EVALUES(2),EVECTOR_2,ERR,ERROR,*999)
+!      CALL Eigenvector(C_e,EVALUES(3),EVECTOR_3,ERR,ERROR,*999)
+!      CALL MATRIX_PRODUCT(F_e_T,F_e,C_e,ERR,ERROR,*999)
+!      CALL DSYEV('V','U',3,C_e,3,EVALUES,WORK,-1,ERR)
+!      IF(ERR.NE.0) WRITE(*,*) "Error in Eigenvalue computation"
+!      LWORK=MIN(LWMAX,INT(WORK(1)))
+!      CALL DSYEV('V','U',3,C_e,3,EVALUES,WORK,LWORK,ERR)
+!      IF(ERR.NE.0) WRITE(*,*) "Error in Eigenvalue computation"
+!      EVECTOR_1=C_e(:,1)
+!      EVECTOR_2=C_e(:,2)
+!      EVECTOR_3=C_e(:,3)
+
+!      DO i=1,3
+!        DO j=1,3
+!          EMATRIX_1(i,j)=EVECTOR_1(i)*EVECTOR_1(j)
+!          EMATRIX_2(i,j)=EVECTOR_2(i)*EVECTOR_2(j)
+!          EMATRIX_3(i,j)=EVECTOR_3(i)*EVECTOR_3(j)
+!        END DO
+!      END DO
+      
+!      CALL MATRIX_TRANSPOSE(F_a_inv,F_a_inv_T,ERR,ERROR,*999)
+
+!      CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_1,TEMP1,ERR,ERROR,*999)
+!      CALL MATRIX_PRODUCT(TEMP1,F_a_inv_T,TEMP1,ERR,ERROR,*999)
+          
+!      CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_2,TEMP2,ERR,ERROR,*999)
+!      CALL MATRIX_PRODUCT(TEMP2,F_a_inv_T,TEMP2,ERR,ERROR,*999)
+
+!      CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_3,TEMP3,ERR,ERROR,*999)
+!      CALL MATRIX_PRODUCT(TEMP3,F_a_inv_T,TEMP3,ERR,ERROR,*999)
+
+      PIOLA_TENSOR=0.0_DP
+      DO i=1,3
+        PIOLA_TENSOR=PIOLA_TENSOR+ &
+          & C(i)*EVALUES(1)**(C(i+3)/2.0_DP-1.0_DP)*N1+ &
+          & C(i)*EVALUES(2)**(C(i+3)/2.0_DP-1.0_DP)*N2+ &
+          & C(i)*EVALUES(3)**(C(i+3)/2.0_DP-1.0_DP)*N3
+      END DO
+      PIOLA_TENSOR=PIOLA_TENSOR*C(7)+2.0_DP*P*AZU
+      
+
+    CASE(EQUATIONS_SET_MULTISCALE_ACTIVE_STRAIN_SUBTYPE)
+
```

Comment:
Multiscale is quite generic when used in this context. It could mean a number of things. Is there a more specific term that might be more appropriate?